lib/io/QuadSinkHdt.ts

```diff
@@ -1,85 +1,123 @@
-  private readonly deleteSourceFiles: boolean;
-  private readonly errorFileDockerRfdhdt: WriteStream;
-  private readonly poolSize: number;
+  private readonly generateIndexes: boolean;
+  private readonly conversionProgress: boolean;
+  private readonly removeSourceFiles: boolean;
+  private readonly conversionConcurrency: number;
+
+  private readonly docker: Docker;
+  private readonly filesToConvert: Set<string>;
 
-  public constructor(
-    options: IQuadSinkFileOptions,
-    poolSize = 1,
-    deleteSourceFiles = false,
-    errorFileDockerRfdhdt = './error_log_docker_rfdhdt.txt',
-  ) {
+  public constructor(options: IQuadSinkHdtOptions) {
     super(options);
-    this.deleteSourceFiles = deleteSourceFiles;
-    this.errorFileDockerRfdhdt = createWriteStream(errorFileDockerRfdhdt);
-    this.poolSize = poolSize;
+    if (!HDTCPP_FORMATS.has(options.outputFormat)) {
+      throw new Error(`Unsupported HDT output format ${options.outputFormat}`);
+    }
+    this.generateIndexes = options.generateIndexes ?? false;
+    this.conversionProgress = options.conversionProgress ?? false;
+    this.removeSourceFiles = options.removeSourceFiles ?? true;
+    this.conversionConcurrency = options.conversionConcurrency ?? 1;
+    this.docker = new Docker();
+    this.filesToConvert = new Set();
   }
 
-  public async push(iri: string, quad: RDF.Quad): Promise<void> {
-    const path = Path.join('./', this.getFilePath(iri));
-    await super.push(iri, quad);
+  protected getFilePath(iri: string): string {
+    const path = super.getFilePath(iri);
+    this.filesToConvert.add(resolve(path));
+    return path;
+  }
+
+  protected async pullDockerImage(): Promise<void> {
+    await new Promise<void>((resolve, reject) => this.docker.pull(HDTCPP_DOCKER_IMAGE, {}, (error, result) => {
+      if (error) {
+        reject(error);
+      }
+      this.docker.modem.followProgress(result!, error => error ? reject(error) : resolve());
+    }));
+  }
 
-    // Add files with the defined extension to the list to be transformed
-    if (this.fileExtension !== undefined && path.includes(this.fileExtension)) {
-      this.files.add(path);
+  protected async convertSingleFile(rdfFilePath: string): Promise<void> {
+    const cmd = [
+      'rdf2hdt',
+      ...this.generateIndexes ? [ '-i' ] : [],
+      ...this.conversionProgress ? [ '-p' ] : [],
+      `${HDTCPP_MOUNT_PATH}/${basename(rdfFilePath)}`,
+      `${HDTCPP_MOUNT_PATH}/${basename(rdfFilePath).replace(this.fileExtension ?? '', '')}.hdt`,
+    ];
+    const createOptions: Docker.ContainerCreateOptions = {
+      // eslint-disable-next-line ts/naming-convention
+      HostConfig: {
+        // eslint-disable-next-line ts/naming-convention
+        AutoRemove: true,
+        // eslint-disable-next-line ts/naming-convention
+        Binds: [ `${dirname(rdfFilePath)}:${HDTCPP_MOUNT_PATH}:rw` ],
+      },
+    };
+    await this.docker.run(HDTCPP_DOCKER_IMAGE, cmd, process.stdout, createOptions);
```

Comment:
I think that since you pipe the docker output to `process.stdout` it "pollutes" the console with the conversion of information of each file. That is why I piped it to a file so that if there is an error the user can check the file and get the latest output to try to fix it.

Reply:
This is why I moved the `-p` flag behind a parameter that is disabled by default, so that the user can choose to print that information if they want to. Currently, it does not get printed by default and the console remains clean.

Reply:
If it always make the error known in someways regardless of the `-p` then I have no problem.